packages/babel-plugin-emotion/src/index.js

```diff
@@ -328,62 +366,83 @@ export default function(babel) {
+                  ) {
                     path.node.arguments.push(
-                      t.stringLiteral(`label:${identifierName.trim()};`)
+                      t.stringLiteral(addSourceMaps(path.node.loc.start, state))
                     )
                   }
-                }
               }
-              // eslint-disable-next-line no-fallthrough
-              case state.importedNames.injectGlobal:
-              case state.importedNames.fontFace:
-                if (
-                  state.opts.sourceMap === true &&
-                  path.node.loc !== undefined
-                ) {
-                  path.node.arguments.push(
-                    t.stringLiteral(addSourceMaps(path.node.loc.start, state))
-                  )
-                }
             }
-          }
 
-          if (
-            (t.isCallExpression(path.node.callee) &&
-              path.node.callee.callee.name === state.importedNames.styled) ||
-            (t.isMemberExpression(path.node.callee) &&
-              t.isIdentifier(path.node.callee.object) &&
-              path.node.callee.object.name === state.importedNames.styled)
-          ) {
-            const identifier = t.isCallExpression(path.node.callee)
-              ? path.node.callee.callee
-              : path.node.callee.object
-            path.replaceWith(
-              buildStyledObjectCallExpression(path, state, identifier, t)
-            )
+            if (
+              (t.isCallExpression(path.node.callee) &&
+                path.node.callee.callee.name === state.importedNames.styled) ||
+              (t.isMemberExpression(path.node.callee) &&
+                t.isIdentifier(path.node.callee.object) &&
+                path.node.callee.object.name === state.importedNames.styled)
+            ) {
+              const identifier = t.isCallExpression(path.node.callee)
+                ? path.node.callee.callee
+                : path.node.callee.object
+              path.replaceWith(
+                buildStyledObjectCallExpression(path, state, identifier, t)
+              )
 
-            if (state.opts.hoist) {
-              hoistPureArgs(path)
+              if (state.opts.hoist) {
+                hoistPureArgs(path)
+              }
             }
+          } catch (e) {
+            throw path.buildCodeFrameError(e)
           }
-        } catch (e) {
-          throw path.buildCodeFrameError(e)
-        }
 
-        path[visited] = true
+          path[visited] = true
+        },
+        exit(path, state) {
+          try {
+            if (
+              path.node.callee &&
+              path.node.callee.property &&
+              path.node.callee.property.name === 'withComponent'
+            ) {
+              if (path.node.arguments.length === 1) {
+                path.node.arguments.push(
+                  t.objectExpression([
+                    buildTargetObjectProperty(path, state, t)
+                  ])
+                )
```

Comment:
Is there a reason this is done on the exit?

Reply:
It seemed like a safe thing to do, since it's tagging on some additional stuff once the other processing is complete. I can move it to `enter` though if you'd prefer.

Reply:
>it's tagging on some additional stuff once the other processing is complete

I'm not sure what you mean by this though it shouldn't affect anything in this case so you can just leave it there.